app/components/Settings/components/JavaManager/JavaArguments.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Icon, Slider, Tooltip, Input, Button, message } from 'antd';
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import store from '../../../../localStore';
+import { promisify } from 'util';
+import { connect } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Constants from '../../../../constants';
+import { Arg } from '../../../../actions/settings';
+import { DATAPATH, WINDOWS } from '../../../../constants';
+
+function JavaArguments(props) {
+  const [globalArg, setglobalArg] = useState();
+
+  const dosName =
+    os.release().substr(0, 2) === 10
+      ? '"-Dos.name=Windows 10" -Dos.version=10.0 '
+      : '';
+
+
+  let defaultARGS = `-Dfml.ignorePatchDiscrepancies=true -Dfml.ignoreInvalidMinecraftCertificates=true ${dosName}${
+    os.platform() === WINDOWS
+      ? '-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump'
+      : ''
+    }-Xms256m -Xmx${props.ram}m`;
+
+  async function readJArgFile() {
+
+    if (store.has('settings.java.javaArg')) {
+      const JavaArgumentsStoreSec = store.get('settings.java.javaArg');
+      setglobalArg(JavaArgumentsStoreSec);
+      props.Arg(JavaArgumentsStoreSec);
+    } else {
+      const JavaArgumentsStore = store.set('settings.java.javaArg', defaultARGS);
+      const JavaArgumentsStoreTh = store.get('settings.java.javaArg');
+      setglobalArg(JavaArgumentsStoreTh);
+      props.Arg(JavaArgumentsStoreTh);
+    }
+
+
+  }
+
+  useEffect(() => {
+    readJArgFile();
+  }, []);
+
+  async function submit() {
+    props.Arg(globalArg);
+    if (globalArg) {
+      store.set('settings.java.javaArg', globalArg);
+    } else message.error("enter valid arguments");
+  }
+
+  function reset() {
+    const JavaArgumentsStore = store.set('settings.java.javaArg', defaultARGS);
+    const JavaArgumentsStoreTh = store.get('settings.java.javaArg');
```

Comment:
Variables names should be camel-cased
Name too generic for a variable / difficult to understand